src/commonMain/kotlin/io/github/koalaplot/core/bar/CandleStickPlotEntry.kt

```diff
@@ -0,0 +1,218 @@
+ *
+ * @param X The type of the x-axis values
+ * @param Y The type of the y-axis values
+ */
+public interface CandleStickPlotEntry<X, Y> {
+    /**
+     * The x-axis value of this [CandleStickPlotEntry].
+     */
+    public val x: X
+
+    /**
+     * The opening price of the candle.
+     */
+    public val open: Y
+
+    /**
+     * The closing price of the candle.
+     */
+    public val close: Y
+
+    /**
+     * The highest price of the candle.
+     */
+    public val high: Y
+
+    /**
+     * The lowest price of the candle.
+     */
+    public val low: Y
+}
+
+/**
+ * Returns an instance of a [CandleStickPlotEntry] using the provided data.
+ */
+public fun <X, Y> candleStickPlotEntry(x: X, open: Y, close: Y, high: Y, low: Y): CandleStickPlotEntry<X, Y> =
+    object : CandleStickPlotEntry<X, Y> {
+        override val x = x
+        override val open = open
+        override val close = close
+        override val high = high
+        override val low = low
+    }
+
+public const val WICKWIDTH: Float = 0.1f
+
+
+@ExperimentalKoalaPlotApi
+@Composable
+public fun <X, Y : Comparable<Y>> XYGraphScope<X, Y>.CandleStickPlot(
+    defaultCandle: @Composable BarScope.(entry: CandleStickPlotEntry<X, Y>) -> Unit = { _ -> },
+    modifier: Modifier = Modifier,
+    candleWidth: Float = 0.5f,
+    wickWidth: Float = WICKWIDTH,
+    animationSpec: AnimationSpec<Float> = KoalaPlotTheme.animationSpec,
+    content: CandleStickPlotScope<X, Y>.() -> Unit
+) {
+    val scope = remember(content, defaultCandle) { CandleStickPlotScopeImpl<X, Y>(defaultCandle) }
+    val data = remember(scope) {
+        scope.content()
+        scope.data
+    }
+
+    val candleBodyEntries = data.map { entry ->
+        verticalBarPlotEntry(entry.x, entry.open, entry.close)
+    }
+
+    val candleWickEntries = data.map { entry ->
+        verticalBarPlotEntry(entry.x, entry.low, entry.high)
+    }
+
+     VerticalBarPlot(
+        data = candleBodyEntries,
+        modifier = modifier,
+        barWidth = candleWidth,
+        animationSpec = animationSpec,
+        bar = { index ->
+            val entry = data[index]
+            val candleColor = if (entry.close >= entry.open) Color.Green else Color.Red
+            DefaultCandleBody(
+                color = candleColor,
+                hoverElement = {
+                    scope.candleContent.invoke(this, entry)
+                }
+            )
+        }
+    )
+
+    VerticalBarPlot(
+        data = candleWickEntries,
+        modifier = modifier,
+        barWidth = wickWidth,
+        animationSpec = animationSpec,
+        bar = { index ->
+            val entry = data[index]
+            val wickColor = if (entry.close >= entry.open) Color.Green else Color.Red
+            DefaultCandleWick(
+                color = wickColor,
+                width = wickWidth,
+                hoverElement = {
+                    scope.candleContent.invoke(this, entry)
```

Comment:
If the goal here is to be able to hover over the wick, then one option is to make the entire candle+wick mark a Composable, passed into the ```CandleStickPlot``` function. Then make a default implementation that combines a Candle Composable with a LineStyle for the wick, that accepts a ```hoverableElement``` modifier.